### DISCUSSION TITLE
Pass param to .qmd in quarto_render

### DISCUSSION THREAD

**Discussion by adpadi** [quarto-dev/quarto-cli#1747] [Q&A] [answered]
Using RStudio, I am trying to pass a parameter to my quarto script and run from a separate script and/or from command line. This worked well in rmarkown using the render and params argument, but argument is not found by the quarto script.

I get the following error message:
```
Quitting from lines 32-83 (report_template.qmd)
Error in eval(expr, envir, enclos) : object 's' not found

```

My approach is:
```
Report_dir<-'C:\\Users\\folder'
scripts<-'C:\\Users\\scripts'
sites = c('site-1', 'site-2')
overwrite=FALSE

for(s in sites){
  report_filename<-file.path(Report_dir, paste0(s, ".docx"))
  #check to see if report is already done
  if (overwrite == FALSE & file.exists(report_filename)){
    next
    } else{
      quarto_render(input = file.path(scripts, "report_template.qmd"),
         output_file=report_filename,
         execute_params=list(site=s))  # is execute_params correct?
}
}
```

Based on the [quarto documentation](https://quarto.org/docs/computations/parameters.html), it appears I need to place the params argument in the YAML, which I tried (although this seemed to be more geared toward Jupyter). I also tried running the quarto file from command line and passing the parameter that way.
`quarto render "C:\Users\folder\report_template.qmd" -P site:'site-1'`

**Comment by mcanouil** [comment, marked as answer]
The issue seems to be in your quarto document rather than the way you are using parameters.

You can try the following small example:

- `qmd`:
  ```
  ---
  title: "My Document"
  params:
    alpha: 0.1
    ratio: 0.1
  ---
  
  # This is alpha = `r params$alpha`
  
  ratio = `r params$ratio`.
  ```

- `r`:
  ```r
  quarto::quarto_render(
    input = "gur.qmd",
    execute_params = list(ratio = 2)
  )
  ```

<img width="1624" alt="Screenshot of VSCode showing in the editor on the left an r script and a quarto document (described previously) and on the right the rendered HTML showing a parameter value overwritten using execute_params argument of quarto_render function" src="https://user-images.githubusercontent.com/8896044/183545676-fb009eb4-a18f-4aa4-8376-4efbd8a29d72.png">

**Reply by adpadi** [reply]
This was helpful. I simply wasn't accessing the parameter using the param$alpha syntax. In Rmarkdown (and e.g. python file from command line) the parameter is simply passed and included as variable. Whereas here, it's stored in the param list, which makes sense, I just didn't notice it was done this way.
